File: tools/photo-album-generator/photo-album.py
import os

num = 0

print open("header.html").read()

for fileName in os.listdir("."):
#	print fileName
	if ".jpg" not in fileName:
#		print "not jpeg"
		continue
	descFile = fileName.replace(".jpg", ".txt")
	desc = ""
	try:
		df = open(descFile)
		desc = df.read().strip()
	except IOError:
		pass
	imgClass = "album-left"
	if num == 1:
		imgClass = "album-right"
		
	print '<img class="' + imgClass + '" src="/flpic/' + fileName + '" alt="' + fileName + '"></img>'
	print '<p class="' + imgClass + '">'
	if num == 0:
		print '<img class="album-arrow-left" src="/flpic/misc/arrow-left-small.png" alt="arrow left" title="' + fileName + '"></img>'	
	if num == 1:
		print '<img class="album-arrow-right" src="/flpic/misc/arrow-right-small.png" alt="arrow right" title="' + fileName + '"></img>'	

	print desc 
	print '</p>'
	print ''
	if True or num == 1:
		print '<div style="clear:both;"></div>'
		print ''

	num = (num + 1) % 2

print "</div>"

File: site/engine/plugins/dollar/photoalbum.php
<?php
    $album_page_prefix = "${full_content_dir}cms/pages/$pageid";
    $cl = $_0;
    $name = $_1;
    $album_text = $_2;
    $album_img_name = "$album_page_prefix/$name.jpg";
?>
<div class="photoalbum-block">
    <img
        class="photoalbum-<?php echo $cl; ?>"
        src="<?php echo $album_img_name; ?>"
        alt="<?php echo $name; ?>"
        title="<?php echo $name; ?>"
    ></img>
    <p class="photoalbum-<?php echo $cl; ?>">
    <?php echo $album_text; ?></p>
</div>
<div class="photoalbum-div" style="clear: both;"></div>
